src/openms/include/OpenMS/ANALYSIS/OPENSWATH/OpenSwathHelper.h

```diff
@@ -181,6 +181,34 @@ namespace OpenMS
     */
     static std::pair<double,double> estimateRTRange(const OpenSwath::LightTargetedExperiment & exp);
 
+    /**
+     * @brief Sample a subset of peptides uniformly across the RT range.
+     *
+     * Splits the RT span (min→max) into @p bins and randomly picks up to
+     * @p peptides_per_bin compounds from each bin. Useful for on-the-fly
+     * iRT calibration without external .irt files.
+     *
+     * @param[in] exp               Full LightTargetedExperiment (the input peptide query parameter assay list for targeted extraction)
+     * @param[in] bins              Number of retention‐time bins (i.e. 10 bins across the RT range for linear iRT, 500-1000 bins across the RT for nonlinear iRT)
+     * @param[in] peptides_per_bin  How many peptides to draw per bin (i.e. 5 peptides for linear iRT, 25 - 50 for non-linear iRT)
+     * @param[in] seed              If non‐zero, used to seed the RNG (deterministic).
+     *                              If zero, will use std::random_device for non-deterministic.
+     * @param[in] filter_decoys     Whether to exclude decoys from the sampled targeted experiment. (You likely do not want to sample with decoys if you are using this method for generating iRTs)
```

Comment:
Is this needed? Would it ever make sense to keep the decoys in the filtering?

Reply:
I initially created the `sampleExperiment` method with the idea for it to be reusable for other contexts, asides for sampling the TargetedExperiment for iRTs. i.e. sampling the PQP for smaller test libraries. I currently convert the PQP to a TSV, and then sample the TSV in R/python, and then convert back to PQP. I can remove the optional `filter_decoys` flag, and enforce the filter for targets only,  if you think it's not necessary.

Reply:
> If you think you would use it in c++/pyopenms feel free to keep it but if there is no current functionality, than I would remove it.

kk, I've removed the optional `filter_decoys` flag, and just explicitly exclude decoys from being sampled for the iRTs.